### PR TITLE
fix: deleting drafts with attachments

### PR DESCRIPTION
**Pull request by opqdonut**
# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [ ] Link to issue

## Documentation
- [x] Update changelog if necessary

## Testing
- [x] Valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

File: test/clj/rems/api/test_applications.clj [hunk 1]
```diff
@@ -562,6 +577,39 @@
...
+      (let [form-id (test-helpers/create-form! {:form/fields [{:field/id "att"
+                                                               :field/type :attachment
+                                                               :field/title {:en "x" :fi "x" :sv "x"}
+                                                               :field/optional false}]})
+            cat-id (test-helpers/create-catalogue-item! {:form-id form-id})
+            app-id (test-helpers/create-application! {:catalogue-item-ids [cat-id]
+                                                      :actor applicant})
+            att-id (-> (request :post (str "/api/applications/add-attachment?application-id=" app-id))
+                       (assoc :params {"file" filecontent})
+                       (assoc :multipart-params {"file" filecontent})
```
Comment (Macroz) [inline test/clj/rems/api/test_applications.clj:590]: Do you know why we need both params and mutipart params?

File: test/clj/rems/api/test_applications.clj [hunk 2]
```diff
@@ -562,6 +577,39 @@
...
+                           :application/permissions
+                           set)
+                       "application.command/delete"))
+        (is (= {:success true}
+               (api-call :post "/api/applications/delete" {:application-id app-id}
+                         api-key applicant)))
+        (is (response-is-not-found?
+             (api-response :get (str "/api/applications/" app-id) nil
+                           api-key applicant)))
+        (is (response-is-not-found?
```
Comment (Macroz) [inline test/clj/rems/api/test_applications.clj:610]: Is this duplicated?

**Review by Macroz — APPROVED** [review]
Some points but otherwise good fix.

File: test/clj/rems/api/test_applications.clj (hunk 1 above)
Comment (opqdonut) [reply test/clj/rems/api/test_applications.clj:590]: No, I copypasted them from the attachment tests :( I'll investigate a bit.

File: test/clj/rems/api/test_applications.clj (hunk 2 above)
Comment (opqdonut) [reply test/clj/rems/api/test_applications.clj:610]: Oh right, I meant to check that the _attachment_ is not found, I'll fix it.

File: test/clj/rems/api/test_applications.clj (hunk 1 above)
Comment (opqdonut) [reply test/clj/rems/api/test_applications.clj:589]: Yeah, they can just be dropped it seems :)

**Pull request merged**